### ISSUE TITLE
autounmount

### ISSUE THREAD

**Issue by pepa65** [vgough/encfs#62]
Does the autounmount-feature still work??
-i $minutes
Somehow it doesn't seem to do that anymore here, but it does accept the -i parameter.

**Comment by Aikhjarto** [comment]
According to the man page, the -i parameter still exists, but IMHO the man page is a bit confusing about this topic:
```
       -i, --idle=MINUTES                                                                                                                                                                       
           Enable automatic unmount of the filesystem after a period of inactivity.  The
           period is specified in minutes, so the shortest timeout period that can be
           requested is one minute.  EncFS will not automatically unmount if there are
           files open within the filesystem, even if they are open in read-only mode.
           However simply having files open does not count as activity.
```

**Comment by pepa65** [comment]
Does it work for you when you use it?

**Comment by Aikhjarto** [comment]
Yes it does work. In normal mode and in reverse mode.

Maybe you can check with lsof or fuser which process is accessing your encfs filesystem.

**Comment by pepa65** [comment]
Hmmm. I'm using: $HOME/git/encfs/encfs/encfs -i 2 $STASH $MNT
Nothing is accessing it after mounting. But it doesn't unmount after 2 minutes.
(Version: 1.8, last entry in Changelog from Sun Nov 23 2014)

**Comment by rfjakob** [comment]
Please run encfs with `-f -v` to get the debug output. Here, the output looks like this, and it does unmount after 1 minute.

    $ encfs -v -i 1 -f a b
    [...]
    08:43:03 (main.cpp:732) idle cycle count: 0, timeout after 6
    08:43:13 (main.cpp:732) idle cycle count: 1, timeout after 6
    08:43:23 (main.cpp:732) idle cycle count: 2, timeout after 6
    08:43:33 (main.cpp:732) idle cycle count: 3, timeout after 6
    08:43:43 (main.cpp:732) idle cycle count: 4, timeout after 6
    08:43:53 (main.cpp:732) idle cycle count: 5, timeout after 6
    08:44:03 (main.cpp:761) Unmounting filesystem b due to inactivity

**Comment by pepa65** [comment]
I happened to try with MNT=/mnt (drwxrwxrwx   2 root root) and it worked like it should.
When MNT=$HOME/Private (drwx------  3 user   user) it doesn't work, every minute: (encfs.cpp:613) statfs of $STASH, (DirNode.cpp:641) created FileNode for $STASH, (encfs.cpp:133) getattr $STASH

Changing the permissions to drwxrwxrwx doesn't fix it.

What I find strange, that there are all kinds of issues with $STASH in the debug for the second case, that don't show up at all in the first case. This looks really buggy to me. Maybe it's a version mismatch?

encfsctl info $STASH
Version 6 configuration; created by EncFS 1.7.4 (revision 20100713)
Filesystem cipher: "ssl/blowfish", version 3:0:0 (using 3:0:2)
Filename encoding: "nameio/block", version 3:0:0 (using 3:0:1)
Key Size: 256 bits
Using PBKDF2, with 217479 iterations
Salt Size: 160 bits
Block Size: 4096 bytes
Each file contains 8 byte header with unique IV data.
Filenames encoded using IV chaining mode.
File holes passed through to ciphertext.

**Comment by rfjakob** [comment]
Looks like for $HOME/Private your desktop environment is polling for changes, this is why it won't unmount.

**Comment by pepa65** [comment]
Is it because I've used the same directory in the past for eCryptfs?? I can't think of any other reason why it would be polled for changes. I have found eCryptfs very unsatisfactory, which is why I moved back to encfs, and I really hope your efforts will make it even better.

When I create a new directory $HOME/NEW, it does automatically unmount.

I remembered and removed the /etc/fstab entry for ecryptfs, en lo: the polling stopped on $HOME/Private... [wtf sorry for my ignorance]

I have another question because of running with -f -v that I'll raise in a new issue.
